rasgoql/rasgoql/data/sqlalchemy.py

```diff
@@ -226,44 +226,47 @@ def get_object_details(self, fqtn: str) -> tuple:
             obj_type = result[0].get("kind")
         return obj_exists, is_rasgo_obj, obj_type
 
-    def get_schema(self, fqtn: str, create_sql: str = None) -> List[Tuple[str, str]]:
+    def get_schema(
+        self,
+        fqtn_or_sql: str,
+    ) -> List[Tuple[str, str]]:
         """
-        Return the schema of a table or view
+        Return the schema of a table, view, or select statement
+
         Params:
-        `fqtn`: str:
-            Fully-qualified table name (database.schema.table)
-        `create_sql`: str:
-            A SQL select statement that will create the view. If this param is passed
-            and the fqtn does not already exist, it will be created and profiled based
-            on this statement. The view will be dropped after profiling
+        `fqtn_or_sql`: str:
+            Either a Fully-qualified table name (database.schema.table)
+            or a SQL select statement that will create a view.
         """
-        fqtn = self.magic_fqtn_handler(fqtn, self.default_namespace)
-        database, schema, table = self.parse_fqtn(fqtn)
         query_sql = (
-            f"SELECT * FROM INFORMATION_SCHEMA.TABLES "
-            f"WHERE table_catalog = '{database}' "
-            f"AND table_schema = '{schema}' "
-            f"AND table_name = '{table}'"
+            "SELECT * FROM INFORMATION_SCHEMA.TABLES "
+            "WHERE table_catalog = '{database}' "
+            "AND table_schema = '{schema}' "
+            "AND table_name = '{table}'"
         )
-        response = []
-        try:
-            if self._table_exists(fqtn):
-                query_response = self.execute_query(query_sql, response="dict")
-            elif create_sql:
-                self.create(create_sql, fqtn, table_type="view")
-                query_response = self.execute_query(query_sql, response="dict")
-                self.execute_query(
-                    f"DROP VIEW {schema}.{table}",
-                    response="none",
-                    acknowledge_risk=True,
-                )
-            else:
-                raise TableAccessError(f"Table {fqtn} does not exist or cannot be accessed.")
-            for row in query_response:
-                response.append((row["name"], row["type"]))
-            return response
-        except Exception as e:
-            self._error_handler(e)
+        # Check for SQL
+        if 'select' in fqtn_or_sql.lower() and ' ' in fqtn_or_sql:
```

Comment:
If we're using this logic three places, would create a reusable function instead, so if we need to change how it operates later we don't miss a spot